crates/ruff_python_ast/src/types.rs

```diff
@@ -70,3 +70,15 @@ impl<'a> From<&RefEquality<'a, Expr>> for &'a Expr {
         r.0
     }
 }
+
+impl<'a> From<RefEquality<'a, Stmt>> for &'a Stmt {
+    fn from(r: RefEquality<'a, Stmt>) -> Self {
+        r.0
+    }
+}
+
+impl<'a> From<RefEquality<'a, Expr>> for &'a Expr {
+    fn from(r: RefEquality<'a, Expr>) -> Self {
+        r.0
+    }
+}
```

Comment:
Nit: Is it possible to use a wildcard implementation here or do we intentionally restrict the type to `Stmt` and `Expr`?
```suggestion
impl<'a, T> From<RefEquality<'a, T>> for &'a T {
    fn from(r: RefEquality<'a, T>) -> Self {
        r.0
    }
}
```

Reply:
Apparently not:

```
error[E0210]: type parameter `T` must be covered by another type when it appears before the first local type (`RefEquality<'a, T>`)
  --> crates/ruff_python_ast/src/types.rs:74:10
   |
74 | impl<'a, T> From<RefEquality<'a, T>> for &'a T {
   |          ^ type parameter `T` must be covered by another type when it appears before the first local type (`RefEquality<'a, T>`)
   |
   = note: implementing a foreign trait is only possible if at least one of the types for which it is implemented is local, and no uncovered type parameters appear before that first local type
   = note: in this case, 'before' refers to the following order: `impl<..> ForeignTrait<T1, ..., Tn> for T0`, where `T0` is the first and `Tn` is the last

For more information about this error, try `rustc --explain E0210`.
error: could not compile `ruff_python_ast` due to previous error
```